appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-document/controller/index.js

```diff
@@ -0,0 +1,37 @@
+import {
+	postDocumentUpload,
+	renderDocumentUpload
+} from '#appeals/appeal-documents/appeal-documents.controller.js';
+
+/** @type {import('@pins/express').RequestHandler<{}>}  */
+const get = async (request, response) => {
+	const { currentComment, currentAppeal, session } = request;
+
+	return renderDocumentUpload({
+		request,
+		response,
+		appealDetails: currentAppeal,
+		backButtonUrl: `/appeals-service/appeal-details/${currentAppeal.appealId}/interested-party-comments/${currentComment.id}`,
+		nextPageUrl: `/appeals-service/appeal-details/${currentAppeal.appealId}/interested-party-comments/${currentComment.id}/add-document/redaction-status`,
+		pageHeadingTextOverride: 'Upload supporting document',
+		allowMultipleFiles: false,
+		documentType: session.costsDocumentType
+	});
+};
+
+/**
+ * @type {import('@pins/express/types/express.js').RequestHandler<{}>}
+ */
+const post = async (request, response) => {
+	const { currentAppeal, currentComment } = request;
+
+	await postDocumentUpload({
+		request,
+		response,
+		nextPageUrl: `/appeals-service/appeal-details/${currentAppeal.appealId}/interested-party-comments/${currentComment.id}/add-document/redaction-status`
+	});
+};
+
+export { get as renderAddDocument, post as postAddDocument };
+export { get as renderDateSubmitted, post as postDateSubmitted } from './date-submitted.js';
```

Comment:
Again, I can see why you've done this, as it's nice to avoid the duplication, but I don't personally feel the tradeoff for clarity (and consistency) is worth it here, and would rather these were defined separately as regular controller functions in each of the required files

Reply:
Do you mean don't have index files?